src/backend/src/services/work-packages.services.ts

```diff
@@ -701,4 +702,103 @@ export default class WorkPackagesService {
 
     return workPackageTemplateTransformer(workPackage);
   }
+
+  static async editWorkPackageTemplate(
+    user: User,
+    workPackageTemplateId: string,
+    templateName: string,
+    templateNotes: string,
+    duration: number | undefined,
+    stage: WorkPackageStage | undefined,
+    blockedBy: BlockedByInfo[],
```

Comment:
you'll  need to make a new type, BlockedByInfoCreateArgs, where id is optional, because any new blockedBys won't have an id yet
example of a similar type for reimbursementproducts:
![image](https://github.com/Northeastern-Electric-Racing/FinishLine/assets/29521172/72943c8e-7009-4cac-a7ba-0878b817d7f5)